src/common/cockpitcloserange.c

```diff
@@ -0,0 +1,110 @@
+#include "cockpitcloserange.h"
+
+#include <assert.h>
+#include <dirent.h>
+#include <err.h>
+#include <errno.h>
+#include <stddef.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/resource.h>
+
+typedef struct {
+  int from;
+  int until;
+} CloseRange;
+
+static int
+closefd (void *data,
+         int fd)
+{
+  CloseRange *ca = data;
+  if (fd >= ca->from && fd <= ca->until)
+    {
+      while (close (fd) < 0)
+        {
+          if (errno == EAGAIN || errno == EINTR)
+            continue;
+          if (errno == EBADF || errno == EINVAL)
+            break;
+          warnx ("couldn't close fd: %m");
+          return -1;
+        }
+    }
+
+  return 0;
+}
+
+#ifndef HAVE_FDWALK
```

Comment:
I have no idea on what kind of Linux system we're supposed to have `fdwalk()`, but it's not a part of glibc.  Please drop the conditionalization of this function and the corresponding check for it in `configure.ac`.

Meanwhile, the callback structure here is actually pointless and has always bothered me: merge the struct typedef plus all three functions in this file into a single simple function, with a single parameter, which does one thing: closes all fds above a certain number.

Reply:
I can add simplification of fdwalk to my TODO list, but I don't want to do it in this PR -- this cleanly moves code around withouth changing behaviour or implementation.

Reply:
I did add a commit which drops the configure check, this is easy enough and safe.

Reply:
OK, on second thought I also added a commit right here which simplifies the function as you mentioned.